### ISSUE TITLE
Reproducing challenge baseline metrics from this repository

### ISSUE THREAD

**Issue by unnat** [epic-kitchens/VISOR-HOS#7]
Hi,

There seems to be some incongruity in the [challenge eval repository](https://github.com/epic-kitchens/C7-VISOR-HOS) and this repository. 

1. The released checkpoints and configs in this repository allow for only 2 classes in HOS: hand and object (active or contact).
I can get hand: 90.877 and object: 30.483 for HOS (matching the paper's numbers). However, it's unclear how to reproduce the hand-side numbers, contact, and combined HO metrics from [released checkpoints](https://github.com/epic-kitchens/VISOR-HOS#pre-trained-weights). Particularly, how do we reproduce [the results on the challenge's CodaLab](https://codalab.lisn.upsaclay.fr/competitions/9969?secret_key=415101af-8490-404b-a28c-d6a4fbad4bcd#results). 
[As mentioned in the challenge repo](https://github.com/epic-kitchens/C7-VISOR-HOS#evaluation), how to go from:
```
|--- ref
|    |--- epick_visor_coco_hos/annotations/{train,val}.json
|    |--- epick_visor_coco_handside/annotations/{train,val}.json
|    |--- epick_visor_coco_contact/annotations/{train,val}.json
|    |--- epick_visor_coco_combineHO/annotations/{train,val}.json
```
to give the associated
```
|--- res
|    |--- instances_predictions_hand_obj.pth
|    |--- instances_predictions_handside.pth
|    |--- instances_predictions_contact.pth
|    |--- instances_predictions_combineHO.pth
``` 
2. [Data] While we can create annotations for `epick_visor_coco_hos` ([ref](https://github.com/epic-kitchens/VISOR-HOS/blob/99f0738e75e289c428d73a1bca2e087a1b1b3797/data_preparation/gen_coco_format.py#L538)), `epick_visor_coco_handside` ([ref](https://github.com/epic-kitchens/VISOR-HOS/blob/99f0738e75e289c428d73a1bca2e087a1b1b3797/data_preparation/gen_coco_format_handside_contact.py#L361)), and `epick_visor_coco_contact` ([ref](https://github.com/epic-kitchens/VISOR-HOS/blob/99f0738e75e289c428d73a1bca2e087a1b1b3797/data_preparation/gen_coco_format_handside_contact.py#L366)), what is the procedure to generate `epick_visor_coco_combineHO/annotations/{train,val}.json` from this repository?

**Comment by ddshan** [comment]
Hi,

Sorry about the confusion here. Let me be more clear.
For 2 related to Data. I have pushed the script to generate combineHO labels under the `data_preparation` folder. Please take a look [here.](https://github.com/epic-kitchens/VISOR-HOS/blob/main/data_preparation/gen_coco_format_combineHO.py)

For 1,  
- The files under `ref` are the COCO-format annotation of VISOR data that you can use the corresponding scripts in the `data_preparation`  folder to generate them. 
As you mentioned, you we can create annotations for epick_visor_coco_hos ([ref](https://github.com/epic-kitchens/VISOR-HOS/blob/99f0738e75e289c428d73a1bca2e087a1b1b3797/data_preparation/gen_coco_format.py#L538)), epick_visor_coco_handside ([ref](https://github.com/epic-kitchens/VISOR-HOS/blob/99f0738e75e289c428d73a1bca2e087a1b1b3797/data_preparation/gen_coco_format_handside_contact.py#L361)), and epick_visor_coco_contact ([ref](https://github.com/epic-kitchens/VISOR-HOS/blob/99f0738e75e289c428d73a1bca2e087a1b1b3797/data_preparation/gen_coco_format_handside_contact.py#L366)), and epick_visor_coco_combineHO with newly added script from ([ref](https://github.com/epic-kitchens/VISOR-HOS/blob/main/data_preparation/gen_coco_format_combineHO.py#L396))
- The files under `res` are automatically saved detection results at `OUTPUT_DIR/inference/instances_predictions.pth` as you specified using the [eval script](https://github.com/epic-kitchens/VISOR-HOS/blob/main/eval.py#L87). Don't forget to add the suffix to the filename `instances_predictions_{task}.pth (task in ['hand_obj', 'handside', 'contact', 'combineHO'])` to distinguish each evaluation task. 

You can evaluate your method on the Validation set on your side by
1. Generate COCO-format annotation for the 4 tasks as mentioned above to get `ref`.
2. Run the evaluation script for the 4 tasks as mentioned above to get `res`.
3. Then by running the [HOS Challenge Evaluation](https://github.com/epic-kitchens/C7-VISOR-HoS) with the `ref` and `res`, you can get the numbers as in the [results table of HOS Challenge Codalab](https://codalab.lisn.upsaclay.fr/competitions/9969?secret_key=415101af-8490-404b-a28c-d6a4fbad4bcd#results) on the Validation set since the annotation for Test set is not public available.

To submit to Codalab,
1. You need to prepare the 4 `PTH` results files for the 4 tasks on Test set in the same format as for the Validation set, in the order as in  [image_id.json](https://github.com/epic-kitchens/C7-VISOR-HOS/blob/main/image_id.json). More details can be find [HOS Challenge Submission Format](https://codalab.lisn.upsaclay.fr/competitions/9969#learn_the_details-submission-format).

Let me know if you have any other questions or anything I can help with!
